thorlcr/activities/external/thexternalslave.ipp

```diff
@@ -0,0 +1,34 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _THEXTERNALSLAVE_IPP
+#define _THEXTERNALSLAVE_IPP
+
+#include "platform.h"
+
+#include "jmutex.hpp"
+#include "jthread.hpp"
+
+#include "slave.ipp"
+#include "thactivityutil.ipp"
+
+
+activityslaves_decl CActivityBase *createExternalSlave(CGraphElementBase *container);
+activityslaves_decl CActivityBase *createExternalSinkSlave(CGraphElementBase *container);
```

Comment:
because no one likes .ipp's and only here to include these factory methods and these are internal to this dll (activityslaves), I tend not to add a activity .ipp for activities and instead these declarations appear near top of slave.cpp (around line 244 onward)